source/geod24/LocalRest.d

```diff
@@ -390,131 +655,175 @@ public final class RemoteAPI (API) : API
-                        },
-                        (FilterAPI filter_api) {
-                            control.filter = filter_api;
-                        },
-                        (Response res) {
-                            if (!isSleeping())
-                                handle(res);
-                            else if (!control.drop)
-                                await_msgs ~= Variant(res);
-                        },
-                        (Command cmd)
+                    Message msg;
+                    if (thisLocalTransceiver.tryReceive(&msg))
+                    {
+                        switch (msg.tag)
                         {
-                            if (!isSleeping())
-                                handle(cmd);
-                            else if (!control.drop)
-                                await_msgs ~= Variant(cmd);
-                        });
+                            case MessageType.request :
+                                if (!isSleeping())
+                                    handleCmd(msg.cmd);
+                                else if (!control.drop)
+                                    await_req ~= msg.cmd;
+                                else if (control.send_response_msg)
+                                    await_drop_req ~= msg.cmd;
+                                break;
+
+                            case MessageType.response :
+                                auto c = thisScheduler.newCondition();
+                                foreach (_; 0..10)
+                                {
+                                    if (thisLocalWaitingManager.exist(msg.res.id))
+                                        break;
+                                    c.wait(10.msecs);
+                                }
+                                if (!isSleeping())
+                                    handleRes(msg.res);
+                                else if (!control.drop)
+                                    await_res ~= msg.res;
+                                break;
+
+                            case MessageType.filter :
+                                control.filter = msg.filter;
+                                break;
+
+                            case MessageType.time_command :
+                                control.sleep_until = Clock.currTime + msg.time.dur;
+                                control.drop = msg.time.drop;
+                                control.send_response_msg = msg.time.send_response_msg;
+                                break;
+
+                            case MessageType.shutdown_command :
+                                terminate = true;
+                                throw new OwnerTerminated();
+
+                            default :
+                                assert(0, "Unexpected type: " ~ msg.tag);
+                        }
+                    }
+
+                    if (thisScheduler !is null)
```

Comment:
This can never be `false`.

Reply:
I tried many things to solve the exception. So this is how it happened.
I'll revise it. :grinning: